dataset/dataset-import.sh

```diff
@@ -1,92 +1,130 @@
+        ;;
+      e)
+        EVENTS_COUNT=$OPTARG
+        ;;
+      o)
+        SESSIONS_COUNT=$OPTARG
+        ;;
+      i)
+        HASH_ITERATIONS=$OPTARG
+        ;;
+      p)
+        REALM_PREFIX=$OPTARG
+        ;;
+      l)
+        KEYCLOAK_URI=$OPTARG
+        ;;
+      t)
+        STATUS_TIMEOUT=$OPTARG
+        ;;
+      ?)
+        echo "Invalid option: $OPT, read the usage carefully -> "
+        help
+        exit 1
+    esac
+  done
+}
 
+create_realms () {
+  echo "Creating $1 realm/s with $2 client/s and $3 user/s with $4 password hash iterations."
+  execute_command "create-realms?count=$1&clients-per-realm=$2&users-per-realm=$3&password-hash-iterations=$4"
 }
 
 create_clients () {
-  KEYCLOAK_URI="${4:-"https://keycloak.$(minikube ip).nip.io"}/realms/master/dataset/"
   echo "Creating $1 client/s in realm $2"
-  check_dataset_status
-  curl -ks "${KEYCLOAK_URI}create-clients?count=$1&realm-name=$2"
+  execute_command "create-clients?count=$1&realm-name=$2"
 }
 
 create_users () {
-  KEYCLOAK_URI="${4:-"https://keycloak.$(minikube ip).nip.io"}/realms/master/dataset/"
   echo "Creating $1 user/s in realm $2"
-  check_dataset_status
-  curl -ks "${KEYCLOAK_URI}create-users?count=$1&realm-name=$2"
+  execute_command "create-users?count=$1&realm-name=$2"
 }
 
 create_events () {
-  KEYCLOAK_URI="${4:-"https://keycloak.$(minikube ip).nip.io"}/realms/master/dataset/"
   echo "Creating $1 event/s in realm $2"
-  check_dataset_status
-  curl -ks "${KEYCLOAK_URI}create-events?count=$1&realm-name=$2"
+  execute_command "create-events?count=$1&realm-name=$2"
 }
 
 create_offline_sessions () {
-  KEYCLOAK_URI="${4:-"https://keycloak.$(minikube ip).nip.io"}/realms/master/dataset/"
   echo "Creating $1 offline sessions in realm $2"
-  check_dataset_status
-  curl -ks "${KEYCLOAK_URI}create-offline-sessions?count=$1&realm-name=$2"
+  execute_command "create-offline-sessions?count=$1&realm-name=$2"
 }
 
 delete_realms () {
-  KEYCLOAK_URI="${3:-"https://keycloak.$(minikube ip).nip.io"}/realms/master/dataset/"
   echo "Deleting realm/s with prefix $1"
-  check_dataset_status
-  curl -ks "${KEYCLOAK_URI}remove-realms?remove-all=true&realm-prefix=$1"
+  execute_command "remove-realms?remove-all=true&realm-prefix=$1"
 }
 
-# shellcheck disable=SC2120
 dataset_provider_status () {
-  KEYCLOAK_URI="${1:-"https://keycloak.$(minikube ip).nip.io"}/realms/master/dataset/"
   echo "Dataset provider status"
-  curl -ks "${KEYCLOAK_URI}status"
-  echo ""
+  execute_command "status"
 }
 
 dataset_provider_status_completed () {
-  KEYCLOAK_URI="${2:-"https://keycloak.$(minikube ip).nip.io"}/realms/master/dataset/"
   echo "Dataset provider status of the last completed task"
   t=0
-  STATUS_TIMEOUT=${1:-120}
-  until curl -ks "${KEYCLOAK_URI}status-completed" | grep '"success":"true"' ; do
-    if [[ $t -gt $STATUS_TIMEOUT ]]
+  until $(execute_command "status-completed" | grep '"success":"true"')
+  do
+    if [[ $t -gt $1 ]]
```

Comment:
@lhanusov I seem to see this behavior consistently where after a successful dataset job completion, running the `task dataset-import -- -a status-completed` gives an output of `../../dataset/dataset-import.sh: line 103: {"status":"previous: command not found` and able to consistently reproduce that, even though the status from the `status-completed` api seems to be coming back accurately.

**Full Log:**
```[kakella@kakella minikube]$ task dataset-import -- -a clear-status-completed
task: Task "split" is up to date
task: Task "ipchange" is up to date
task: Task "keycloak-cli-download" is up to date
task: Task "tlsdisableagent" is up to date
task: Task "env" is up to date
task: Task "prometheus" is up to date
task: Task "promtail" is up to date
task: Task "jaeger" is up to date
task: Task "tempo" is up to date
task: Task "loki" is up to date
task: Task "monitoring" is up to date
task: Task "dataset" is up to date
task: Task "datasetprovider" is up to date
task: Task "keycloak" is up to date
task: Task "keycloak-cli-unzip" is up to date
task: Task "gatlinguser" is up to date
[dataset-import] Action: [clear-status-completed] Dataset provider clears the status of the last completed task
[dataset-import] Skipping dataset status check
[dataset-import] 
[kakella@kakella minikube]$ task dataset-import -- -a create-offline-sessions -o 100 -n test-realm
task: Task "split" is up to date
task: Task "ipchange" is up to date
task: Task "keycloak-cli-download" is up to date
task: Task "tlsdisableagent" is up to date
task: Task "env" is up to date
task: Task "prometheus" is up to date
task: Task "promtail" is up to date
task: Task "tempo" is up to date
task: Task "loki" is up to date
task: Task "jaeger" is up to date
task: Task "dataset" is up to date
task: Task "monitoring" is up to date
task: Task "datasetprovider" is up to date
task: Task "keycloak" is up to date
task: Task "keycloak-cli-unzip" is up to date
task: Task "gatlinguser" is up to date
[dataset-import] Action: [create-offline-sessions] Creating 100 offline sessions in realm test-realm
[dataset-import] {"status":"Task started successfully","task":{"startTimeMs":"1673472778200","message":"Creation of 100 offline sessions"},"task-status-url":"https://keycloak.192.168.39.160.nip.io/realms/master/dataset/status"}
[kakella@kakella minikube]$ curl http://keycloak.$(minikube ip).nip.io/realms/master/dataset/status-completed
{"status":"previous task completed","task":{"success":"true","startTimeMs":"1673472778200","message":"Creation of 100 offline sessions","endTimeMs":"1673472779282"}}[katask dataset-import -- -a status-completed
task: Task "split" is up to date
task: Task "ipchange" is up to date
task: Task "tlsdisableagent" is up to date
task: Task "keycloak-cli-download" is up to date
task: Task "env" is up to date
task: Task "prometheus" is up to date
task: Task "tempo" is up to date
task: Task "loki" is up to date
task: Task "promtail" is up to date
task: Task "jaeger" is up to date
task: Task "monitoring" is up to date
task: Task "dataset" is up to date
task: Task "datasetprovider" is up to date
task: Task "keycloak" is up to date
task: Task "keycloak-cli-unzip" is up to date
task: Task "gatlinguser" is up to date
[dataset-import] Action: [status-completed] Dataset provider status of the last completed task
[dataset-import] ../../dataset/dataset-import.sh: line 103: {"status":"previous: command not found
[dataset-import] Polling...1s
[dataset-import] ../../dataset/dataset-import.sh: line 103: {"status":"previous: command not found
[dataset-import] Polling...2s
[dataset-import] ../../dataset/dataset-import.sh: line 103: {"status":"previous: command not found
^Ctask: Signal received: "interrupt"
```